### ISSUE TITLE
failure to load libraries closes MATLAB

### ISSUE THREAD

**Issue by skaterzero807** [aphysci/gravity#180]
TLDR: MATLAB closes when Java fails to load native libraries such as libzmq

## Steps to reproduce:
Using gravity 1.16 in MATLAB 2019a (also repeatable with gravity 1.13 in 2016b and 2018b)
1. Make sure Gravity isn't already installed in MATLAB (i.e. no java class path or library path is set)
2. Run MATLAB with the "-logfile" option to dump its command output to a text file
3. Try to create a Gravity node within MATLAB:
```
n = com.aphysci.gravity.swig.GravityNode;
```

### Issue 1
MATLAB closes, and the log file will have the Java error:

```
Native code library failed to load. Tried both gravity_wrap and libgravity_wrap.
java.lang.UnsatisfiedLinkError: D:\MATLAB\Temp Test Files\gravity crash debug\bin\libgravity_wrap.dll: The specified procedure could not be found
```

### Issue 2
If the Gravity installation is valid, no error will occur and Step 3 will succeed. Clear "n" and try Step 3 again and MATLAB closes, now the log file will have the error:
```
Native code library failed to load. Tried both zmq and libzmq.
java.lang.UnsatisfiedLinkError: Native Library D:\MATLAB\Temp Test Files\gravity crash debug\deps\libzmq.dll already loaded in another classloader
```

Both issues seem tied to System.exit [L50 and L61 in gravity.i](http://github.com/aphysci/gravity/blob/2cbf889ff5587e55ecccffbf1f027aec0cca6eb8/src/api/java/src/swig/gravity.i#L50)

see [this answer](https://www.mathworks.com/matlabcentral/answers/93828-why-does-matlab-exit-when-i-close-a-java-gui-application-that-was-invoked-from-within-matlab) from Mathworks that explains the behavior

## Possible fixes (not sure if these will break other implementations):
1. Don't call System.exit and instead throw an unhandled exception ([see this SO question](https://stackoverflow.com/questions/20946862/terminating-a-java-program-without-exit-matlab))
2. Don't try to load the library if it's already loaded? Or don't load the library at all and let the user setup the Java library path? Not sure what's the right path, I couldn't make sense of [this SO answer](https://stackoverflow.com/questions/36936948/java-lang-unsatisfiedlinkerror-native-library-xxx-so-already-loaded-in-another)
